app/controllers/application_controller.rb

```diff
@@ -44,4 +44,15 @@ def store_location
       session[:return_to] = request.original_fullpath
     end
   end
+
+  def store_referer
```

Comment:
Well, one is the path the user requested and the other is the location they came from...